write-fonts/src/tables/variations/ivs_builder.rs

```diff
@@ -119,8 +119,27 @@ impl VariationStoreBuilder {
         *self.all_regions.entry(region).or_insert(next_idx)
     }
 
+    fn get_axis_count(&self) -> u16 {
+        let axis_count = self
+            .all_regions
+            .keys()
+            .next()
+            .map(|reg| reg.region_axes.len());
+        match axis_count {
+            Some(count) => count.try_into().unwrap(),
+            None => {
+                // is it ever reasonable to have an IVS builder to which you add
+                // no items? it seems possible? but maybe zero is valid in that
+                // case, since it would also mean we have an empty region list?
+                // Help me Cosimo!
+                0
+            }
```

Comment:
at a glance this looks like reinvention of unwrap or default?

What would it mean if the regions had inconsistent axis count? Do we validate for that?

Reply:
We should be able to test this right? Assert the axis count when there are no regions is 0?

Reply:
yes, this is mostly an excuse for me to write this note.


the spec requires the axis count to match between fvar and the RegionList, and so it's possible that other code in the ecosystem is depending on that being the case. In the particular case of this value being zero, if we ever hit this branch I think we are generating an inconsistent font, and we would currently hit the debug assert if we tried to access the region list; however in release we would just receive an empty set of regions, which would be fine.


we don't currently have a mechanism to validate things across table boundaries, so this is not checked. If we are going to merge this patch, we will want to add that check; I don't think we should be compiling this.

Reply:
> We should be able to test this right? Assert the axis count when there are no regions is 0?

the issue is that if there are no region there still may be axes, we just don't have any way of knowing what they are.

Reply:
I mean literally just to test get_axis_count, e.g. `assert_eq!(0, VariationStoreBuilder::new().get_axis_count())`

Reply:
> this is mostly an excuse for me to write this note

the note belongs in an issue imho :)

Reply:
Basically we need to decide whether it is ever okay to generate an `ItemVariationStore` with no deltas. If it's not okay, then this patch should go in, and we should add that validation check, or make the builder return a result or something. If it *is* okay than we need to make the user tell us how many axes there are when they create the builder.

Reply:
Unless there is something in the spec that bans empty IVS I think it's OK. An odd choice certainly but not illegal.

Reply:
> whether it is ever okay to generate an ItemVariationStore with no deltas

no deltas (i.e. ItemVarData without items) is ok even though it's a bit weird, why bother to have the varstore to begin with, most of the time when building one we check if it turns out to be empty, and don't bother storing one (it's usually optional, if it isn't optional then the parent table likely is optional as a whole so both are omitted).

However, having no axes at all is _not_ ok for an allegedly "variable" font. No fvar axes (the var region axis count == len(fvar.axes) by definitino) means the font is not variable! How can something that has "variation" in its very name actually define no axes of variation?!